adam-core/src/main/scala/org/bdgenomics/adam/converters/VariantContextConverter.scala

```diff
@@ -339,6 +339,373 @@ private[adam] class VariantContextConverter extends Serializable with Logging {
     ???
   }
 
+  // variant conversion functions
+
+  private[converters] def formatNames(
+    vc: HtsjdkVariantContext,
+    vb: Variant.Builder): Variant.Builder = {
+
+    splitIds(vc).foreach(vb.setNames(_))
+    vb
+  }
+
+  private[converters] def formatFilters(
+    vc: HtsjdkVariantContext,
+    vb: Variant.Builder): Variant.Builder = {
+
+    vb.setFiltersApplied(vc.filtersWereApplied)
+    if (vc.filtersWereApplied) {
+      vb.setFiltersPassed(!vc.isFiltered)
+    }
+    if (vc.isFiltered) {
+      vb.setFiltersFailed(new java.util.ArrayList(vc.getFilters));
+    }
+    vb
+  }
+
+  private[converters] def formatSomatic(
+    vc: HtsjdkVariantContext,
+    vb: Variant.Builder): Variant.Builder = {
+
+    Option(vc.getAttribute("SOMATIC").asInstanceOf[java.lang.Boolean]).foreach(vb.setSomatic(_))
+    vb
+  }
+
+  private val variantConversionFns: Iterable[(HtsjdkVariantContext, Variant.Builder) => Variant.Builder] = Iterable(
+    formatNames(_, _),
+    formatFilters(_, _),
+    formatSomatic(_, _)
+  )
+
+  private[converters] def extractNames(
+    v: Variant,
+    vcb: VariantContextBuilder): VariantContextBuilder = {
+
+    joinNames(v) match {
```

Comment:
[Idiomatic scala](http://blog.originate.com/blog/2014/06/15/idiomatic-scala-your-options-do-not-match/) demands a `fold` here and not a `match`.